omod/src/main/webapp/resources/scripts/patientsearch/patientSearchWidget.js

```diff
@@ -240,55 +240,60 @@ function PatientSearchWidget(configuration){
-                if(age == '' && widgetBirthdate != ''){
-                    age = formatAge(widgetBirthdate);
+        var results = removeDuplicates(results);
+        searchResultsData = searchResultsData.concat(results);
+        _.each(results, function(patient) {
+        	var birthdate = '';
+            var widgetBirthdate = patient.person.birthdate;
+            if(patient.person.birthdate){
+            	birthdate = moment(patient.person.birthdate).format(configuration.dateFormat);
+                if( patient.person.birthdateEstimated ){
+                	birthdate = "~ "+birthdate;
+                }else{
+                	birthdate = "&nbsp;&nbsp; "+birthdate;
                 }
-                var dataRow = [identifier, patient.person.personName.display, patient.person.gender,
-                    age, birthdate];
-                jq.each(attributeTypes, function(index, typeName){
-                    var attributeValue = "";
-                    jq.each(patient.attributes, function(index, attribute) {
-                        var attrType = attribute.attributeType;
-                        if (attrType != null && !attribute.voided && typeName == attrType.name) {
-                            if(attribute.value != null) {
-                                if(!attribute.value.display) {
-                                    attributeValue = attribute.value;
-                                }else{
-                                    attributeValue = attribute.value.display;
-                                }
-                            }
-                            return false;
-                        }
-                    });
-                    dataRow.push(attributeValue);
-                });
-                dataRows.push(dataRow);
-            });
-        }else if(config.initialPatients){
-            //show the recently viewed
-            searchResultsData = initialPatientData;
-            dataRows = initialData;
-        }
+            }
+            var identifier = patient.patientIdentifier != null ? patient.patientIdentifier.identifier : null;
+            if(_.contains(initialPatientUuids, patient.uuid)){
+            	identifier = patient.patientIdentifier.identifier+
+                	" <span class='recent-lozenge'>"+config.messages.recent+"</span>";
+            }
+            var age = patient.person.age;
+            if(age == '' && widgetBirthdate != ''){
+            	age = formatAge(widgetBirthdate);
+            }
+                
+            // apply filters
+            var matchedPatient = true;
+            if(jq("#filter-age").val()) {
+            	matchedPatient = age == jq("#filter-age").val();
+           	} else if(jq("#filter-gender").val()) {
+           		matchedPatient = patient.person.gender == jq("#filter-gender").val();
+           	}
+            	
+            if(matchedPatient) {
+	        	var dataRow = [identifier, patient.person.personName.display, patient.person.gender,
```

Comment:
There are some formatting issues after this lines. Could you look in to this as well?